File: 0392-is-subsequence/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
    var S,T int
    S=len(s)
    T=len(t)
    if  S>T{
        return false
    }
    if S==0{
        return true
    }
    var j int 
    j=0
    for i:=0;i<T;i++{
        if t[i]==s[j]{
            if j==S-1{
                return true
            }
            j++
        }
    }
    return false 

    
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        def helper(i,j):
            if i==len(s):
                return True
            #iska matlab ki t katam ho gaya hai  but  s baki hai 
            if j==len(t):
                return False
            if s[i]==t[j]:
                return helper(i+1,j+1)
            else:
                return helper(i,j+1)
        return helper(0,0)                







        